serviceWorker.js

```diff
@@ -0,0 +1,167 @@
+if (typeof window === 'undefined') {
```

Comment:
Wrap this code in smaller functions, these big if/else at the top level if difficult to reason about